01/ryanh153/wordvalue.py

```diff
@@ -0,0 +1,31 @@
+from data import DICTIONARY, LETTER_SCORES
+
+"""Load dictionary into a list and return list"""
+def load_words():
+
+    with open(DICTIONARY,'r') as dicFile:
+        dictionary = dicFile.readlines()
+
+    dictionary = [word.strip() for word in dictionary]
+
+    return dictionary
+
+
+"""Calculate the value of the word entered into function
+using imported constant mapping LETTER_SCORES"""
+def calc_word_value(word):
+    return sum([LETTER_SCORES[char] for char in word.upper() if char in LETTER_SCORES.keys()])
+
+
+"""Calculate the word with the max value, can receive a list
+of words as arg, if none provided uses default DICTIONARY"""
+def max_word_value(words=None):
+    if words == None:
+        words = load_words()
+
+    scores = [calc_word_value(word) for word in words]
+
+    return words[scores.index(max(scores))]
```

Comment:
ha nice! This is also correct but much harder than it has to be. You first search for the maximum, than its position. Max needs to sort the list, otherwise it cannot know the maximum element. If you sort the list already, than you know where the maximum is, dont you? Its at the end. So another way to achieve the same is `sorted(words)[-1]` because sorted sorts ascending for default.